File: benchmarks/java/Hashmap.java
// benchmarks/java/Hashmap.java
import java.util.HashMap;

public class Hashmap {
    static String keyOf(int i) {
        long h = (long)i * 2654435761L % 1000000L;
        return "key_" + h;
    }
    public static void main(String[] args) {
        final int N = 500000;
        HashMap<String, Long> m = new HashMap<>(N * 2);
        for (int i = 0; i < N; i++) m.put(keyOf(i), (long)i);

        long total = 0;
        for (int i = 0; i < N; i++) {
            Long v = m.get(keyOf(i));
            if (v != null) total += v;
        }
        System.out.println(total);
        System.out.println(m.size());
    }
}

File: benchmarks/cpp/hashmap.cpp
// benchmarks/cpp/hashmap.cpp
#include <cstdio>
#include <string>
#include <unordered_map>

static std::string key_of(int i) {
    long long h = (long long)i * 2654435761LL % 1000000LL;
    return "key_" + std::to_string(h);
}

int main() {
    const int N = 500000;
    std::unordered_map<std::string, long long> m;
    m.reserve((size_t)N * 2);
    for (int i = 0; i < N; ++i) m[key_of(i)] = i;

    long long total = 0;
    for (int i = 0; i < N; ++i) {
        auto it = m.find(key_of(i));
        if (it != m.end()) total += it->second;
    }
    std::printf("%lld\n%zu\n", total, m.size());
    return 0;
}